### DISCUSSION TITLE
Scatter and ignore root node

### DISCUSSION THREAD

**Discussion by kssalanio** [mpi4py/mpi4py#22] [Q&A] [answered]
Hi, I have a question regarding scatter() and scatterv().

I want to scatter a list of items, but ignore the root node in that scattering. Is that what "MPI_IN_PLACE" is for?
If so, how is the syntax usage for it in mpi4py? I can only seem to find examples for it in C.

Also, does the same apply for gather() and gatherv()?

**Comment by dalcinl** [comment, marked as answer]
Your working is a bit confusing, I'm not sure what's your definition of "ignore" in this context.

Are you asking about pickle-based communication (**lowercase** methods)? In that case no, the root has to provide its own entry to be sent to self, but it is trivial to just pass `None` in the position corresponding to root, and then scatter.

If you are asking about buffer-based communication (**uppercase** methods), then the answer is maybe. Again, it depends what you understand for ignore. Please check the MPI standard document, the use of `MPI_IN_PLACE` is clearly defined in there. About how to use `MPI_IN_PLACE` mpi4py, you just pass `MPI.IN_PLACE` as the send or recv buffer of the collective call, e.g. `comm.Gather(recvbuf=MPI.IN_PLACE, ...)`.

**Reply by kssalanio** [reply]
My question was answered by the information you provided, thank you very much! 

But for the sake of clarification here is what I meant by "ignore". Say for example I have clusters with processors rank 0-4, root=0, and a list [a, b, c, d, e, f, g, h] . I want to send data only to ranks 1,2,3,4 so:

rank 0 gets None or []
rank 1 gets [a, b]
rank 2 gets [c, d]
rank 3 gets [e, f]
rank 4 gets [g, h]

And yes, I was asking for both pickle-based and buffer-based communication. Thanks again!
